packages/cli/src/ui/components/InputPrompt.tsx

```diff
@@ -234,8 +245,95 @@ export const InputPrompt: React.FC<InputPromptProps> = ({
 
   const handleInput = useCallback(
     (key: Key) => {
+      // Terminal drag-and-drop focus event sequences
+      const FOCUS_EVENT_IN_EDITOR = '\u001b[I'; // When editor has focus
+      const FOCUS_EVENT_OUT_OF_EDITOR = "'"; // When editor does NOT have focus (macOS Finder.app)
+
+      // Helper function to reset drag collection state
+      const resetDragState = () => {
+        if (dragTimeoutRef.current) {
+          clearTimeout(dragTimeoutRef.current);
+          dragTimeoutRef.current = null;
+        }
+        setIsDragCollecting(false);
+        isDragCollectingRef.current = false;
+        setDragBuffer('');
+      };
+
+      // Check if this is a drag-and-drop focus event from any terminal
+      const isDragFocusEvent = (sequence: string) => {
+        return (
+          sequence === FOCUS_EVENT_IN_EDITOR ||
+          sequence === FOCUS_EVENT_OUT_OF_EDITOR
+        );
+      };
+
+      // Handle normal paste operations first
+      if (key.paste) {
+        if (isDragCollectingRef.current) {
+          resetDragState();
+        }
+        // Continue with normal paste handling
+      }
+
+      // Detect focus event that might start a drag operation (only if NOT a paste)
+      if (isDragFocusEvent(key.sequence) && !key.paste && !shellModeActive) {
+        setIsDragCollecting(true);
+        isDragCollectingRef.current = true;
+        setDragBuffer('');
+
+        if (dragTimeoutRef.current) {
+          clearTimeout(dragTimeoutRef.current);
+        }
+        dragTimeoutRef.current = setTimeout(() => {
+          resetDragState();
+        }, DRAG_START_TIMEOUT_MS);
+
+        return;
+      }
+
+      // Collect individual characters during drag operation
+      if (
+        isDragCollectingRef.current &&
+        key.sequence &&
+        key.sequence.length === 1 &&
+        !key.paste
+      ) {
+        setDragBuffer((prev) => {
+          const newBuffer = prev + key.sequence;
+
+          // Reset collection timeout for each character
+          if (dragTimeoutRef.current) {
+            clearTimeout(dragTimeoutRef.current);
+          }
+
+          // Set completion timeout - longer delay for long paths
```

Comment:
is there really a longer delay for longer paths? doesn't seem like it from the code. would generally recommend a single somewhat longer delay that trying to optimize the delay as this code path should not trigger outside of drag and drop